domain/relation/state/relation.go

```diff
@@ -899,6 +900,10 @@ WHERE relation_unit_uuid = $getRelationUnitEndpointName.relation_unit_uuid`, arg
 
 // GetRelationUnit retrieves the UUID of a relation unit based on the given
 // relation UUID and unit name.
+//
+// The following error types can be expected to be returned:
+//   - [relationerrors.RelationUnitNotFound] if the relation unit cannot be
```

Comment:
As a flyby, I changed this function to return `RelationUnitNotFound` instead of `UnitNotFound` as I was using its subsidiary function. I've checked all the callsites and none of them check for this error, so no further changes were required outside of the relation package.